packages/stack-ui/src/components/brand-icons.tsx

```
@@ -0,0 +1,215 @@
+    </svg>
+  );
+}
+
+export function X({ iconSize } : { iconSize: number} ) {
+  return (
+    <svg aria-label="X" viewBox="0 0 1200 1227" width={iconSize} height={iconSize} xmlns="http://www.w3.org/2000/svg">
+      <path fill="#FFFFFF" d="M714.163 519.284L1160.89 0H1055.03L667.137 450.887L357.328 0H0L468.492 681.821L0 1226.37H105.866L515.491 750.218L842.672 1226.37H1200L714.137 519.284H714.163ZM569.165 687.828L521.697 619.934L144.011 79.6944H306.615L611.412 515.685L658.88 583.579L1055.08 1150.3H892.476L569.165 687.854V687.828Z"/>
+    </svg>
+  );
+}
+
+export function Mapping({
+  provider,
+  iconSize,
+}: {
+  provider: string,
+  iconSize: number,
+}) {
+  switch (provider) {
+    case "google": {
+      return <Google iconSize={iconSize} />;
+    }
+    case "github": {
+      return <GitHub iconSize={iconSize} />;
+    }
+    case "facebook": {
+      return <Facebook iconSize={iconSize} />;
+    }
+    case "microsoft": {
+      return <Microsoft iconSize={iconSize} />;
+    }
+    case "spotify": {
+      return <Spotify iconSize={iconSize} />;
+    }
+    case "discord": {
+      return <Discord iconSize={iconSize} />;
+    }
+    case "gitlab": {
+      return <Gitlab iconSize={iconSize} />;
+    }
+    case "bitbucket": {
+      return <Bitbucket iconSize={iconSize} />;
+    }
+    case "linkedin": {
+      return <LinkedIn iconSize={iconSize} />;
+    }
+    case "apple": {
+      return <Apple iconSize={iconSize} />;
+    }
+    case "x": {
+      return <X iconSize={iconSize} />;
+    }
+    default: {
+      return <></>;
```

Whenever you can, throw errors instead of showing empty or loading values

```suggestion
      return throw new StackAssertionError(`Icon not found for provider: ${provider}`);
```

---

Done